### DISCUSSION TITLE
Mesh adaptation in fluid mechanics using Feel++

### DISCUSSION THREAD

**Discussion by prudhomm** [feelpp/feelpp#1550] [Portfolio]
We are currently developing a fluid framework in moving domains handling rigid and elastic bodies, including articulated swimmers like the 3-spheres swimmer.

Here are two videos that use the mesh adaptation library mmg to deal with bad quality mesh 

- [Moving sphere](https://youtu.be/Kr88UjglF9E) 
- [Moving Sphere in a Driven Cavity ](https://youtu.be/Kr88UjglF9E)

**Comment by Trophime** [comment]
@prudhomm Can we consider anistropic mesh adaption like what we did with Cecile based on Loseuille work (Gradient recovery)?
